core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -240,6 +240,21 @@ private void removeDelayMarkers(RoutingAllocation allocation) {
     }
 
     /**
+     * Reset failed allocation counter for unassigned shards
+     */
+    private void resetFailedAllocationCounter(RoutingAllocation allocation) {
+        final RoutingNodes.UnassignedShards.UnassignedIterator unassignedIterator = allocation.routingNodes().unassigned().iterator();
+        while (unassignedIterator.hasNext()) {
+            ShardRouting shardRouting = unassignedIterator.next();
+            UnassignedInfo unassignedInfo = shardRouting.unassignedInfo();
+            unassignedIterator.updateUnassigned(new UnassignedInfo(UnassignedInfo.Reason.MANUAL_ALLOCATION, unassignedInfo.getMessage(),
```

Comment:
Hi @bleskes , I was waiting for your review to discuss this. I introduce the new state in order to avoid this failing assertion:

     java.lang.AssertionError: failedAllocations: 0 for reason ALLOCATION_FAILED

	at __randomizedtesting.SeedInfo.seed([AA218FFC2E9D0C23:C056D02DF261AAF8]:0)
	at org.elasticsearch.cluster.routing.UnassignedInfo.<init>(UnassignedInfo.java:248)
	at org.elasticsearch.cluster.routing.allocation.AllocationService.resetFailedAllocationCounter(AllocationService.java:253)
	at org.elasticsearch.cluster.routing.allocation.AllocationService.reroute(AllocationService.java:292)
	at org.elasticsearch.cluster.routing.allocation.MaxRetryAllocationDeciderTests.testSingleRetryOnIgnore(MaxRetryAllocationDeciderTests.java:117)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)

Reply:
Fair enough. Can we just touch the shards whose failed allocation acounter > 0 - there might be other reasons to have an unassigned shard.